### DISCUSSION TITLE
[Setup] Github Access Token for Xcode

### DISCUSSION THREAD

**Discussion by sabarrag** [CS342/.github#7] [Setup] [answered]
### Description

Hi! I remember from lecture that as part of the setup for connecting Github and Xcode was creating a personal access token when cloning the the CS 342-2023-Application using the SSH link. I wanted to know what specific scopes we should select when creating this token is it just "repo" or are there other scopes we should include?

### Reproduction

N/A

### Expected behavior

N/A

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
Hi @sabarrag, thank you for using the CS342 GitHub Discussions for your question!

You can find out more information about the requirements for the personal access tokens (PATs) for Xcode when you add an account. The sheet will tell you all the required scopes:
<img width="400" alt="Screenshot 2023-01-20 at 6 12 20 PM" src="https://user-images.githubusercontent.com/28656495/213839470-83cac712-7622-4531-912e-09db3bb8a193.png">
You can find more information about [adding a remote repository account to Xcode in the Xcode documentation](https://help.apple.com/xcode/mac/11.4/index.html?localePath=en.lproj#/deve306e855d).

If you want to add SSH-based authentication to push any code from your Mac (inside Xcode but also outside of that using the command line), you can follow the GitHub Documentation about [Adding a new SSH key to your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) and their linked steps for more information and a step-by-step instruction.

**Reply by sabarrag** [reply]
Okay, thank you!

**Reply by PSchmiedmayer** [reply]
@sabarrag You're welcome!
Let us know if that solves your problem. If so, please answer the comment as an answer so we can mark the discussion as solved and others easily know where to find the answer 👍
